content/cpp/concepts/constructors/constructors.md

```diff
@@ -16,75 +16,121 @@ CatalogContent:
 
-### Syntax and Examples
+## Syntax
 
-A constructor method is defined within a class by using the `ClassName` followed by parentheses.
+A constructor method is defined within a class by using the `ClassName` followed by parentheses. It does not have a return type.
+
+```cpp
+class  ClassName {
+  public:
+    ClassName() {
+      // Constructor code
+    }
+};
+```
+
+## Examples
+
+- Basic Constructor
+
+A basic constructor, also known as a default constructor, is a special type of constructor in C++ that takes no parameters. The compiler will automatically provide a default constructor if there is no constructor explicitly defined for a class. This constructor is used for initializing an object when no specific values are needed, and it also initializes member variables to default values.
 
 ```cpp
 class MyClass {
   public:
-    // Define a constructor that outputs a message
-    MyClass() {
-      std::cout << "The constructor was executed!";
+    // Define a basic constructor
+    MyClass () {
+      std::cout << "The constructor was executed!" << std::endl;
     }
 };
 
 int main() {
-  MyClass myObj;  // This calls the constructor
+  MyClass myObject;
   return 0;
 }
 ```
 
-Using arguments in a constructor:
+The result of the following code:
 
 ```cpp
-class Circle {
+The constructor was executed!
+```
+
+- Parameterized Constructor
+  A parameterized constructor is one that accepts arguments, allowing initial values to be set for an object's attributes at the time of creation. This constructor is useful for initializing objects with custom data or values that differ between instances.
+
+```cpp
+class Rectangle {
   public:
-    int X;
-    int Y;
-    int radius;
+    int width;
+    int height;
 
-    Circle (int a, int b, int c) {
-      X = a;
-      Y = b;
-      radius = c;
+    // Declare the constructor
+    Rectangle (int w, int h) {
+      width = w;
+      height = h;
     }
 };
 
 int main() {
-  // Call the constructor using arguments
-  Circle myCircle1(5,5,10);
-  Circle myCircle2(0,0,5);
+  // Declare implicitly
+  Rectangle rect1(10, 20);
+  // Declare explicitly
+  Rectangle rect2 = Rectangle(5, 15);
+
+  std::cout << "Rectangle 1: Width = " << rect1.width << ", Height = " << rect1.height << std::endl;
+  std::cout << "Rectangle 2: Width = " << rect2.width << ", Height = " << rect2.height << std::endl;
+}
+```
+
+The result of the following code:
 
-  // Print out the attribute values set for each object
+```cpp
+Rectangle 1: Width = 10, Height = 20
+Rectangle 2: Width = 5, Height = 15
+```
 
-  std::cout << "X=" << myCircle1.X << ", Y=" << myCircle1.Y << ", radius=" << myCircle1.radius << "\n";
-  // Output: X=5, Y=5, radius=10
+## Codebyte Example
 
-  std::cout << "X=" << myCircle2.X << ", Y=" << myCircle2.Y << ", radius=" << myCircle2.radius << "\n";
-  // Output: X=0, Y=0, radius=5
+Run the codebyte example below to see how constructors work:
```

Comment:
Codebyte examples should be preceded by a description of what the example does, for example, in this case, it can be something like this - 

```
The following example defines a simple class called Circle that has a single public attribute, radius, and a default constructor that initializes this attribute.
```